### PR TITLE
Add order origin tracking.

### PR DESCRIPTION
**Pull request by budzanowski**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Add orders origin tracking to the tracker snapshot data. The orders section in the snapshot will now have the `created_via` section which will list all distinct values for orders `_created_via` meta with the total count for each value.

It will allow us to at least get some indication of how many orders are happening ‘off-site’ and could be an indicator towards how many merchants sell ‘off’ of their WooCommerce site.

### How to test the changes in this Pull Request:

1. Using an WC installation that has orders data ( one can use smooth generator )
2. Trigger snapshot generation using WP CLI `wp wc tracker snapshot --format=json`
3. Observe the orders section
image

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> run changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/includes/class-wc-tracker.php [hunk 1]
```diff
@@ -542,6 +543,39 @@ private static function get_orders_by_gateway() {
...
+		global $wpdb;
+
+		$orders_origin = $wpdb->get_results(
+			"
+			SELECT
+				meta_value as origin, COUNT( DISTINCT ( orders.id ) ) as count
+			FROM
+				wp_posts orders
+			LEFT JOIN
+				wp_postmeta order_meta ON order_meta.post_id = orders.id
```
Comment (barryhughes) [inline plugins/woocommerce/includes/class-wc-tracker.php]: We should avoid hard-coding the table prefix (within some installations a custom prefix may be defined). How about:

```suggestion
				$wpdb->posts orders
			LEFT JOIN
				$wpdb->postmeta order_meta ON order_meta.post_id = orders.id
```

File: plugins/woocommerce/includes/class-wc-tracker.php [hunk 2]
```diff
@@ -542,6 +543,39 @@ private static function get_orders_by_gateway() {
 		return $orders_by_gateway_currency;
 	}
 
+	/**
+	 * Get orders origin details.
+	 *
```
Comment (barryhughes) [inline plugins/woocommerce/includes/class-wc-tracker.php]: Minor/coding standards (no trailing spaces):

```suggestion
	 *
```

File: plugins/woocommerce/includes/class-wc-tracker.php [hunk 3]
```diff
@@ -542,6 +543,39 @@ private static function get_orders_by_gateway() {
...
+				wp_postmeta order_meta ON order_meta.post_id = orders.id
+			WHERE
+				meta_key = '_created_via'
+			GROUP BY
+				meta_value;
+			"
+		);
+
+		$orders_by_origin = array();
+		foreach( $orders_origin as $origin ) {
```
Comment (barryhughes) [inline plugins/woocommerce/includes/class-wc-tracker.php]: Minor/coding standards:

```suggestion
		foreach ( $orders_origin as $origin ) {
```

File: plugins/woocommerce/includes/class-wc-tracker.php [hunk 4]
```diff
@@ -542,6 +543,39 @@ private static function get_orders_by_gateway() {
...
+			$orders_by_origin[$origin->origin] = (int) $origin->count;
```
Comment (barryhughes) [inline plugins/woocommerce/includes/class-wc-tracker.php]: Minor/coding standards:

```suggestion
			$orders_by_origin[ $origin->origin ] = (int) $origin->count;
```

File: plugins/woocommerce/includes/class-wc-tracker.php [hunk 5]
```diff
@@ -542,6 +543,39 @@ private static function get_orders_by_gateway() {
...
+		}
+
+		return array(
+			'created_via' => $orders_by_origin
+		);
```
Comment (barryhughes) [inline plugins/woocommerce/includes/class-wc-tracker.php]: Minor/coding standards → should be either (trailing comma for each item in a multiline array):

```suggestion
		return array(
			'created_via' => $orders_by_origin,
		);
```

Or else a one-liner (trailing commas not required):

```suggestion
		return array( 'created_via' => $orders_by_origin );
```

**Review by barryhughes — CHANGES_REQUESTED** [review]
Left some notes we should address. Beyond that, tested with the suggested correction to the query and the expected data was available via `wp wc tracker snapshot --format=json`.

**Comment by barryhughes** [comment]
✍🏽 Depending on how quickly the changes can be made, we may need a code freeze exception to ship this in 7.1.

**Comment by budzanowski** [comment]
@barryhughes thank you very much for the suggestions. I have used them directly.

Tested locally - everything works.

**Review by barryhughes — APPROVED** [review]
Thanks, LGTM 🚀 

Will keep an eye on the checks and merge once they complete :+1:

**Pull request merged**